internal/bminventory/inventory_v2_handlers.go

```diff
@@ -311,7 +316,36 @@ func (b *bareMetalInventory) V2CompleteInstallation(ctx context.Context, params
 		return common.GenerateErrorResponder(err)
 	}
 
-	if !*params.CompletionParams.IsSuccess {
+	parse := func(data interface{}) ([]models.OperatorMonitorReport, error) {
+		raw, err := json.Marshal(data)
+		if err != nil {
+			return nil, err
+		}
+		var result []models.OperatorMonitorReport
+		if err := json.Unmarshal(raw, &result); err != nil {
+			return nil, err
+		}
+		return result, nil
+	}
+
+	if !swag.BoolValue(params.CompletionParams.IsSuccess) {
+		// Check if there is a report on failing operators, and compose a report event.
+		// The event simply shows a list of failed operators' names.
+		if data, ok := params.CompletionParams.Data[clusterOperatorReportKey]; ok {
+			if ops, err := parse(data); err == nil {
+				//filter only the names of failed operators
+				failedOps := funk.Filter(ops, func(op models.OperatorMonitorReport) bool {
+					return op.Status == models.OperatorStatusFailed
```

Comment:
What if we have operators that were slow to rollout? won't they show as `models.OperatorStatusProgressing`?

Reply:
This is an aggregated status that is sent from the controller to the service. Please refer to the controller code for the exact mapping. In this case progressing after timeout will count as a failure as well

Reply:
ok